semgrep/semgrep/app/session.py

```diff
@@ -31,13 +33,43 @@ class UserAgent:
 
     @tags.default
     def get_default_tags(self) -> Set[str]:
+        result = set()
         if os.getenv("SEMGREP_USER_AGENT_APPEND"):
-            return set({os.environ["SEMGREP_USER_AGENT_APPEND"]})
-        return set()
+            result.add(os.environ["SEMGREP_USER_AGENT_APPEND"])
+
+        try:
+            remote_url = subprocess.check_output(
+                ["git", "remote", "get-url", "origin"],
+                cwd=Path(__file__).parent,
+                stderr=subprocess.DEVNULL,
+                encoding="utf-8",
+            ).strip()
+            sha = subprocess.check_output(
+                [
+                    "git",
+                    "describe",
+                    # a --match value never matches will give us SHA instead of most recent tag
+                    "--match=nah_dont_actually_match",
+                    "--always",
+                    "--dirty",
+                ],
+                cwd=Path(__file__).parent,
+                stderr=subprocess.DEVNULL,
+                encoding="utf-8",
+            ).strip()
+            # If we installed semgrep in `.venv/` in the semgrep-docs repo,
+            # this function would report semgrep-docs commit SHAs in the user agent.
+            # This is why we verify the origin URL, and why check with .endswith()
+            if remote_url.replace(".git", "").endswith("/semgrep"):
+                result.add(f"sha/{sha}")
```

Comment:
I thought this'd be handy to be able to recognize when weird metrics come from faulty commits, and to filter out metrics from development in general. But it's highly optional.

Reply:
eh seems special case-y. Is it difficult to filter it out server side?

Reply:
without this we don't have any markers that'd let us know if a metric was sent from one of our devs

Reply:
so it's impossible to filter; but we can also just decide not to care about filtering